dapr/proto/components/v1/secretstore.proto

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+package dapr.proto.components.v1;
+
+import "dapr/proto/components/v1/common.proto";
+
+option go_package = "github.com/dapr/dapr/pkg/proto/components/v1;components";
+
+// Interface for secret store.
+service SecretStore {
+    // Initializes the secret store with the given metadata.
+    rpc Init(SecretStoreInitRequest) returns (SecretStoreInitResponse) {}
+
+    // Returns a list of implemented secret store features.
+    rpc Features(FeaturesRequest) returns (FeaturesResponse) {}
+
+    // Get an individual secret from the store.
+    rpc Get(GetSecretRequest) returns (GetSecretResponse) {}
+
+    // Get all secrets from the store.
+    rpc BulkGet(BulkGetSecretRequest) returns (BulkGetSecretResponse) {}
+
+    // Ping the pubsub. Used for liveness porpuses.
+    rpc Ping(PingRequest) returns (PingResponse) {}
+  }
+
+// Request to initialize the secret store.
+message SecretStoreInitRequest {
+  MetadataRequest metadata = 1;
+}
+
+// Response from initialization.
+message SecretStoreInitResponse {}
+
+// GetSecretRequest is the message to get secret from secret store.
+message GetSecretRequest {
+  // The name of secret key.
+  string key = 2;
```

Comment:
Silly question, but should field numbering generally start at `1`?

Reply:
Yeah, I numbered it with 2 purposefully while trying to match the runtime/proto definition. I think, to make it consistent I will change with 1.